src/Symfony/Component/Notifier/Bridge/Discord/DiscordTransportFactory.php

```diff
@@ -17,27 +17,36 @@
 
 /**
  * @author Mathieu Piot <math.piot@gmail.com>
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
  */
 final class DiscordTransportFactory extends AbstractTransportFactory
 {
-    public function create(Dsn $dsn): DiscordTransport
+    public function create(Dsn $dsn): DiscordTransport|DiscordBotTransport
```

Comment:
I think this should use the interface as return type (as done in the interface) instead of using the implementation classes